cpp/src/io/comp/decompression.cpp

```diff
@@ -757,6 +739,38 @@ std::vector<uint8_t> decompress(compression_type compression, host_span<uint8_t
   CUDF_FAIL("Unsupported compressed stream type");
 }
 
+[[nodiscard]] size_t find_split_index(device_span<device_span<uint8_t const> const> inputs,
```

Comment:
Looks like a duplicate of the `find_split_index()` in compression.cpp, can we use the same function with a `tparam perhaps a boolean enum` to compute the `threshold` and `total_host_to_single_device_thread_ratio`